src/lib.rs

```diff
@@ -21,6 +21,17 @@
 ))]
 #[macro_use] extern crate std;
 
+#[cfg(any(
+    target_os = "android",
+    target_os = "netbsd",
+    target_os = "solaris",
+    target_os = "redox",
+    target_os = "dragonfly",
+    target_os = "haiku",
+    target_os = "emscripten",
+    target_os = "linux",
+))]
+mod utils;
```

Comment:
It seems a bit unnecessary to `cfg` this (and potential point of failure). Perhaps better to hide warnings by renaming it `_utils`?

Reply:
Don't forget that `utils` uses `std::io::Error`.